client/opentracing/src/main/java/io/opentracing/APMSpanBuilder.java

```diff
@@ -34,20 +34,24 @@
 
     private TraceRecorder recorder;
     private ContextSampler sampler;
+    private String serviceName;
+    private String buildStamp;
 
     /**
      * @param operationName The operation name
      * @param recorder The trace recorder
      */
-    APMSpanBuilder(String operationName, TraceRecorder recorder, ContextSampler sampler) {
+    APMSpanBuilder(String operationName, TraceRecorder recorder, ContextSampler sampler, String serviceName, String buildStamp) {
```

Comment:
This is the place where we would need to duplicate storing the `serviceName` and `buildStamp`, something that could be more elegantly solved by wrapping the `TraceRecorder` into a `EnvironmentAwareTraceRecorder`